pallets/map-set/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+//! A pallet that implements a storage set on top of a storage map and demonstrates performance
+//! tradeoffs when using vec sets.
+
+use frame_support::{decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure};
+use frame_system::{self as system, ensure_signed};
+use frame_support::storage::IterableStorageMap;
+use sp_std::prelude::*;
+use sp_std::collections::btree_set::BTreeSet;
+use account_set::AccountSet;
+
+#[cfg(test)]
+mod tests;
+
+/// A maximum number of members. When membership reaches this number, no new members may join.
+pub const MAX_MEMBERS: u32 = 16;
+
+pub trait Trait: system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as VecMap {
+		// The set of all members. The bool value is useless and will always be
+		// true. It is necessary because the underlying storage can't distinguish
+		// between 0-byte values and non-existent values so `()` can't be used.
```

Comment:
```suggestion
		// The set of all members. The bool value is a workaround and will always be
		// `true`. It would be nicer to map to `()`, but `()` is encoded as 0-bytes. The
		// underlying storage cannot distinguish between keys 0-byte values and keys
		// not present in the map.
```

Let me try again

Reply:
I think it's actually even more subtle than this actually. If I remember correctly, the runtime storage works fine mappiong to `()` but querying from off-chain does not. Basically I'm trying to tell the learner that there be dragons when mapping to `()` and they're wise to avoid it.

Reply:
https://github.com/paritytech/substrate/issues/5986

Reply:
> paritytech/substrate#5986

TIL! FWIW this has bitten us in the past in ethereum as well. Easy to get wrong.

Reply:
We should link this in the docs for the recipe :rocket: